subsys/net/ip/net_pkt.c

```diff
@@ -1794,6 +1794,10 @@ static void clone_pkt_attributes(struct net_pkt *pkt, struct net_pkt *clone_pkt)
 
 #if defined(CONFIG_IEEE802154)
 	net_pkt_set_ieee802154_rssi(clone_pkt, net_pkt_ieee802154_rssi(pkt));
+#if defined(CONFIG_IEEE802154_SELECTIVE_TXPOWER)
+	net_pkt_set_ieee802154_txpwr(clone_pkt, net_pkt_ieee802154_txpwr(pkt));
+	net_pkt_set_ieee802154_txpwr_use(clone_pkt, net_pkt_ieee802154_txpwr_use(pkt));
+#endif
```

Comment:
rssi and txpwr use the same memory in union. This code causes the memory to be copied twice.

Reply:
So you'd suggest to have:
```c
        net_pkt_set_ieee802154_rssi(clone_pkt, net_pkt_ieee802154_rssi(pkt));
#if defined(CONFIG_IEEE802154_SELECTIVE_TXPOWER)
        /* ieee802154_rssi and ieee802154_txpwr form a union, copying one of them is enough */
	net_pkt_set_ieee802154_txpwr_use(clone_pkt, net_pkt_ieee802154_txpwr_use(pkt));
#endif
```